keystorage/azure/src/main/java/tech/pegasys/signers/azure/AzureKeyVault.java

```diff
@@ -149,4 +152,14 @@ public <R> MappedResults<R> mapSecrets(final BiFunction<String, String, R> mappe
     }
     return MappedResults.newInstance(result, errorCount.intValue());
   }
+
+  private static boolean secretPropertiesPredicate(
+      final Map<String, String> tags, final SecretProperties secretProperties) {
+    if (tags == null || tags.isEmpty()) {
+      return true; // we don't want to filter if user-supplied tags map is empty
+    }
+
+    return secretProperties.getTags() != null // return false if remote secret doesn't have any tags
+        && secretProperties.getTags().entrySet().containsAll(tags.entrySet());
```

Comment:
This requires matching both the key and value, which is slightly different from what was done for AWS. For AWS it could either be keys, values or both keys and values that would be matched. Don't remember why we did that though.